src/NServiceBus.Persistence.DynamoDB.Tests/Outbox/OutboxPersisterTests.cs

```diff
@@ -209,6 +209,77 @@ public async Task Should_return_record_even_if_metadata_and_transport_operations
         Assert.AreEqual(1, result.TransportOperations.Length);
     }
 
+    [Test]
+    public void Should_throw_when_transport_operations_not_complete()
+    {
+        // There is a metadata record but no transport operations returned
+        // this can happen due to read-committed isolation levels
+        client.QueryRequestResponse = r => new QueryResponse
+        {
+            Items = new List<Dictionary<string, AttributeValue>> {
+                new()
+                {
+                    { "PK", new AttributeValue("OUTBOX#endpointIdentifier#someMessageId")},
+                    { "SK", new AttributeValue("OUTBOX#METADATA#someMessageId")},
+                    { "Dispatched", new AttributeValue { BOOL = false }},
+                    { "OperationsCount", new AttributeValue { N = "1" }}
+                }
+            }
+        };
+
+        var contextBag = new ContextBag();
+
+        var exception = Assert.ThrowsAsync<PartialOutboxResultException>(async () => await persister.Get("someMessageId", contextBag));
+        Assert.That(exception!.Message, Does.Contain("Partial outbox results retrieved with 0 instead of 1 transport operation while attempting to load the outbox records for the message id 'someMessageId'."));
+    }
+
+    [Test]
+    public void Should_throw_when_transport_operations_not_complete_even_when_paging()
+    {
+        // There is a metadata record but no transport operations returned
+        // this can happen due to read-committed isolation levels
+        var responses = new Queue<QueryResponse>();
+        // First response contains metadata entry but no transport operation
+        responses.Enqueue(new QueryResponse
+        {
+            Items = new List<Dictionary<string, AttributeValue>>
+            {
+                {
+                    new()
+                    {
+                        {
+                            "SK", new AttributeValue($"OUTBOX#METADATA#someMessageId")
+                        },
```

Comment:
should we be consistent and also add the PK?
```suggestion
                        { "PK", new AttributeValue("OUTBOX#endpointIdentifier#someMessageId")},
                        { "SK", new AttributeValue($"OUTBOX#METADATA#someMessageId") },
```